src/model/CodeExplorer.tsx

```diff
@@ -0,0 +1,93 @@
+import Editor from '@monaco-editor/react'
+import { useState, useEffect, useCallback, useMemo } from 'react'
+import { useTheme } from '@mui/material/styles'
+import Box from '@mui/material/Box'
+import TreeView from '@mui/lab/TreeView'
+import ExpandMoreIcon from '@mui/icons-material/ExpandMore'
+import ChevronRightIcon from '@mui/icons-material/ChevronRight'
+import Grid from '@mui/material/Grid'
+import TreeItem from '@mui/lab/TreeItem'
+import { useGetVersionFile, useGetVersionFileList } from '@/data/version'
+import { filesArrayToTree, RenderTree } from '@/utils/tree'
+
+export default function CodeExplorer({
+  id,
+  addQueryParameter,
+}: {
+  id: string
+  addQueryParameter: (key: string, value: string) => void
+}) {
+  const [path, setPath] = useState<string | undefined>(undefined)
+  const { fileList } = useGetVersionFileList(id)
+  const { file } = useGetVersionFile(id, path)
+  const theme = useTheme()
+
+  const changePath = useCallback(
+    (newPath: string) => {
+      addQueryParameter('path', newPath)
+      setPath(newPath)
+    },
+    [addQueryParameter]
+  )
+
+  const tree: RenderTree = useMemo(() => {
+    if (!fileList) {
+      return {
+        id: '',
+        name: '',
+      }
+    }
+
+    return filesArrayToTree(fileList.fileList)
+  }, [fileList])
+
+  useEffect(() => {
+    if (path || !fileList?.fileList?.length) {
+      return
+    }
+
+    changePath(fileList.fileList[0].fileName)
+  }, [fileList, changePath, path])
+
+  const renderTree = (nodes: RenderTree) => (
+    <TreeItem
+      key={nodes.id}
+      nodeId={nodes.id}
+      label={nodes.name}
+      onClick={() => {
+        if (!nodes.id.endsWith('/')) {
+          changePath(nodes.id)
+        }
+      }}
+    >
+      {Array.isArray(nodes.children) ? nodes.children.map((node) => renderTree(node)) : null}
+    </TreeItem>
+  )
+
+  return (
+    <Box>
```

Comment:
As we're not using any properties, this can be a fragment `<></>`

Reply:
Actually, can you just be removed altogether, as you've only got one top level component